### DISCUSSION TITLE
How to add pages from webspace in Sulu Admin Bar ?

### DISCUSSION THREAD

**Discussion by niji-rmejean** [sulu/sulu#6920] [Q&A] [answered]
Hi,

Is there a way to make this kind of shortcut easily ? I'm using Sulu 2.4 and I saw no documentation about that. Not sure that's possible ...

![image](https://user-images.githubusercontent.com/102292967/204571671-78a15d5a-8f74-4e73-a2de-2c88b2a7672d.png)

Thanks by advance !

**Comment by alexander-schranz** [comment, marked as answer]
@niji-rmejean there is currently no way to add a specific page to the sulu admin bar. You can just add any view which does not require a route parameter (pages require the page uuid).

All around the Admin extending is documented here:

 - https://docs.sulu.io/en/2.5/book/extend-admin.html

To get step by step into that area we recommend doing the sulu-workshop

https://github.com/sulu/sulu-workshop/pulls

You maybe could even workaround by using default params in some kind of custom view to achieve adding a page to the main nav, you would need UUID as env variable for that. But not sure if possible, and currently that is nothing which is supported out of the box or even recommended by us.

I think a feature like `Bookmarking` some view as a user itself would be quiet interesting. So a user can maybe on a dashboard see its last bookmarks or a menu item bookarks.

What is special about that page that you want to have it outside in the main navigation? Maybe a feature like System Pages could also solve this: https://github.com/sulu/sulu/issues/6832. Let us know more about your specific use case and your specific page what it does handle.

**Reply by niji-rmejean** [reply]
We decided to put our front menu in the webspace to easily get it with SuluHeadlessBundle. 
We also would like to make a shortcut to the homepage admin in the menu bar.

**Reply by alexander-schranz** [reply]
@niji-rmejean the SuluHeadlessBundle has an own endpoint for navigation contexts. A menu don't sounds like something which should be part of the Pages Tree as it is not a page some visitor will be visiting.

You should consider using the normal navigation context and so the Headless Bundle endpoint for it: https://github.com/sulu/suluheadlessbundle#navigation

For all other global data which is not page specific Sulu has something called `Default Snippets` / `Snippet Areas`. Which also have there own endpoint: https://github.com/sulu/suluheadlessbundle#snippet-areas. See also https://docs.sulu.io/en/2.5/cookbook/default-snippets.html. We mostly recommend creating one `settings.xml` snippet here which provides you all data which is managed by the content manager for the base view of your site.
